src/main/java/com/nha/abdm/wrapper/hip/HIPClient.java

```diff
@@ -0,0 +1,24 @@
+/* (C) 2024 */
+package com.nha.abdm.wrapper.hip;
+
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Component;
+import org.springframework.web.reactive.function.client.WebClient;
+
+@Component
+public class HIPClient {
+
+  @Value("${getPatientPath}")
+  private String patientUrl;
+
+  public HIPPatient getPatient(String patientId) {
+    String url = patientUrl + "/" + patientId;
+
+    WebClient.Builder webClientBuilder = WebClient.builder();
+    ResponseEntity<HIPPatient> responseEntity =
```

Comment:
Probably WebClient can be built once and reused for all the requests.

Reply:
Agree, it would have been ideal to have weclient created once with basic configs and use it again and again for different paths. But I dont see that option for now. Please share any reference for ways we can achieve that.